pipeline/transform.py

```diff
@@ -0,0 +1,299 @@
+    country_names = [country["name"].lower() for country in countries.values()]
+    state_names = [state["name"].lower() for state in states.values()]
+    city_names = [city["name"].lower() for city in cities.values()]
+
+    all_locations = country_names + state_names + city_names + country_codes
+
+    return all_locations
+
+
+def config_log() -> None:
+    """
+    Terminal logs configuration.
+    """
+    logging.basicConfig(
+        format="{asctime} - {levelname} - {message}",
+        style="{",
+        datefmt="%Y-%m-%d %H:%M",
+        level=logging.INFO,
+    )
+
+
+def convert_from_unix_to_datetime(unix: str) -> datetime:
+    """
+    Converts a unix timestamp into
+    'DD-MM-YYYY' format.
+    Returns a "None" string if an error occurs.
+    """
+    try:
+        date = datetime.fromtimestamp(int(unix))
+        return date.strftime("%d-%m-%Y")
+    except ValueError as e:
+        logging.error("Invalid unix timestamp: %s", e)
+        return "None"
+
+
+def convert_date_format(date_str: str) -> str:
+    """
+    Takes dates in the "DD B YYYY" format and
+    converts them to 'DD-MM-YYYY'.
+    Returns a "None" string if an error occurs.
+    """
+    try:
+        date = datetime.strptime(date_str, "%d %B %Y")
+        return date.strftime("%d-%m-%Y")
+    except ValueError as e:
+        logging.error("Invalid date format: %s", e)
+        return "None"
+
+
+def convert_to_full_url(url: str) -> str:
+    """
+    Checks if a url is fully formatted.
+    Returns the full url if it is not.
+    Returns the given url if it is.
+    """
+    if url.startswith("http"):
+        return url
+
+    return "https:" + url
+
+
+def validate_album_and_track(item_type: str) -> bool:
+    """
+    Checks whether given item is an album or a track.
+    Returns True if it is, False if it isn't.
+    """
+
+    return item_type in ["a", "t"]
+
+
+def get_sale_information(sales_dict: dict) -> list[dict]:
+    """
+    Returns a list of dictionaries containing the
+    sales information.
+    """
+    sales_info = []
+    events = sales_dict["feed_data"]["events"]
+    locations = get_locations()
+
+    for event in events:
+        items = event["items"][0]
+
+        if not validate_album_and_track(items["item_type"]):
+            continue
+
+        if not items["url"]:
+            release_date = "None"
+            genres = []
+
+        release_date = get_release_date_from_url(items["url"])
+
+        if not release_date:
+            release_date = "None"
+
+        genres = get_genres_from_url(items["url"], locations)
+
+        if not items["album_title"]:
+            items["album_title"] = "None"
+
+        sale_date = convert_from_unix_to_datetime(items["utc_date"])
+
+        event_information = {"item_type": items["item_type"],
+                             "item_description": items["item_description"],
+                             "album_title": items["album_title"],
```

Comment:
Although this is what the API calls it, maybe we could rename 'album_name' to something like 'release_name' here to better reflect capturing both albums and tracks, and better match with the ERD?

Reply:
Great point ^

Reply:
I've changed:
- `item_type` to `release_type` as we're explicitly dealing with album/track drops.
- `item_description` to `release_name` as it better describes what it is in relation to the table.

Since `album_name` is actually the name of an album that is either being sold as a whole or the album that a track is being sold from, I think it's fine to leave that as is.